### PR TITLE
fixing kafka hlc routing table and upgrade kafka version to 0.8.2.2

### PR DESCRIPTION
**Pull request by fx19880617**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -92,7 +92,7 @@
     <avro.version>1.7.6</avro.version>
     <helix.version>0.6.5</helix.version>
     <!-- jfim: for Kafka 0.9.0.0, use zkclient 0.7 -->
-    <kafka.version>0.8.2.0</kafka.version>
+    <kafka.version>0.8.2.2</kafka.version>
```
Comment (jfim) [inline pom.xml]: Is there a specific reason why the Kafka version is upgraded?

**Comment by fx19880617** [comment]
I'm hitting an issue of consumer config initialization. The only workable way is to upgrade kafka version.


This is the stacktrace:
2016/03/20 09:08:29.669 ERROR [HelixStateTransitionHandler] Exception while executing a state transition task testTable_REALTIME_1458464909287_0__ALL__1458464909320
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.helix.messaging.handling.HelixStateTransitionHandler.invoke(HelixStateTransitionHandler.java:344)
	at org.apache.helix.messaging.handling.HelixStateTransitionHandler.handleMessage(HelixStateTransitionHandler.java:290)
	at org.apache.helix.messaging.handling.HelixTask.call(HelixTask.java:84)
	at org.apache.helix.messaging.handling.HelixTask.call(HelixTask.java:41)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoSuchMethodError: scala.Predef$.augmentString(Ljava/lang/String;)Ljava/lang/String;
	at kafka.utils.VerifiableProperties.getIntInRange(VerifiableProperties.scala:75)
	at kafka.utils.VerifiableProperties.getInt(VerifiableProperties.scala:58)
	at kafka.utils.ZKConfig.<init>(ZkUtils.scala:749)
	at kafka.consumer.ConsumerConfig.<init>(ConsumerConfig.scala:90)
	at kafka.consumer.ConsumerConfig.<init>(ConsumerConfig.scala:94)
	at com.linkedin.pinot.core.realtime.impl.kafka.KafkaHighLevelStreamProviderConfig.getKafkaConsumerConfig(KafkaHighLevelStreamProviderConfig.java:142)
	at com.linkedin.pinot.core.realtime.impl.kafka.KafkaConsumerManager.acquireConsumerAndIteratorForConfig(KafkaConsumerManager.java:82)
	at com.linkedin.pinot.core.realtime.impl.kafka.KafkaHighLevelConsumerStreamProvider.start(KafkaHighLevelConsumerStreamProvider.java:64)
	at com.linkedin.pinot.core.data.manager.realtime.RealtimeSegmentDataManager.<init>(RealtimeSegmentDataManager.java:132)
	at com.linkedin.pinot.core.data.manager.realtime.RealtimeTableDataManager.addSegment(RealtimeTableDataManager.java:117)
	at com.linkedin.pinot.server.starter.helix.HelixInstanceDataManager.addSegment(HelixInstanceDataManager.java:195)
	at com.linkedin.pinot.server.starter.helix.SegmentOnlineOfflineStateModelFactory$SegmentOnlineOfflineStateModel.onBecomeOnlineFromOfflineForRealtimeSegment(SegmentOnlineOfflineStateModelFactory.java:125)
	at com.linkedin.pinot.server.starter.helix.SegmentOnlineOfflineStateModelFactory$SegmentOnlineOfflineStateModel.onBecomeOnlineFromOffline(SegmentOnlineOfflineStateModelFactory.java:98)
	... 12 more
2016/03/20 09:08:29.679 ERROR [StateModel] Default rollback method invoked on error. Error Code: ERROR

**Comment by jfim** [comment]
It's unlikely that the change in version from 0.8.2.0 to 0.8.2.2 has anything to do with it though. My guess is that this version of Kafka depends on scala-library 2.10.4 but we added larray recently, which depends on scala-library 2.10.3. Doing a `mvn dependency:tree` shows that the scala-library that's pulled isn't stable (sometimes it gets pulled through larray, sometimes it gets pulled through Kafka), so that's a more likely explanation. For example, the Hadoop part will contain scala-library 2.10.3 while pinot-server apparently contains 2.10.4.

However, you can always run your local build with an arbitrary version of Kafka, by doing `mvn -Dkafka.version=0.8.2.2 -DskipTests install`. I wouldn't put this in master (the Kafka version change) without understanding what's causing this, but the routing table change is a good catch.

Can you update the PR to only keep the routing table change?

**Comment by fx19880617** [comment]
Thanks JF, updated!

**Comment by jfim** [comment]
Just merged it, you should be good.

**Pull request merged**